File: backend/src/main/java/nutricentro/services/NotificationService.java
package nutricentro.services;

import nutricentro.dtos.notifications.NotificationResponseDTO;
import nutricentro.dtos.notifications.NotificationSummaryDTO;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface NotificationService {
    List<NotificationResponseDTO> getNotifications();
    NotificationSummaryDTO getSummary();
    void markAsRead(String notificationKey);
    void markAllAsRead();
}

File: frontend/nutricentro/src/app/core/services/notification-service.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../enviroment/enviroment';
import { NotificationResponseDTO, NotificationSummaryDTO } from '../models/notification-model';

@Injectable({
  providedIn: 'root',
})
export class NotificationService {
  private http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiUrl}/notifications`;

  getNotifications() {
    return this.http.get<NotificationResponseDTO[]>(this.apiUrl);
  }

  getSummary() {
    return this.http.get<NotificationSummaryDTO>(`${this.apiUrl}/summary`);
  }

  markAsRead(notificationKey: string) {
    return this.http.patch<void>(`${this.apiUrl}/${encodeURIComponent(notificationKey)}/read`, {});
  }

  markAllAsRead() {
    return this.http.patch<void>(`${this.apiUrl}/read-all`, {});
  }
}
